beacon_node/network/src/router.rs

```diff
@@ -0,0 +1,537 @@
+        network_send: mpsc::UnboundedSender<NetworkMessage<T::EthSpec>>,
+        executor: task_executor::TaskExecutor,
+        log: slog::Logger,
+    ) -> error::Result<mpsc::UnboundedSender<RouterMessage<T::EthSpec>>> {
+        let message_handler_log = log.new(o!("service"=> "router"));
+        trace!(message_handler_log, "Service starting");
+
+        let (handler_send, handler_recv) = mpsc::unbounded_channel();
+
+        let sync_logger = log.new(o!("service"=> "sync"));
+        let (beacon_processor_send, beacon_processor_receive) =
+            mpsc::channel(MAX_WORK_EVENT_QUEUE_LEN);
+
+        // spawn the sync thread
+        let sync_send = crate::sync::manager::spawn(
+            executor.clone(),
+            beacon_chain.clone(),
+            network_globals.clone(),
+            network_send.clone(),
+            beacon_processor_send.clone(),
+            sync_logger,
+        );
+
+        BeaconProcessor {
+            beacon_chain: Arc::downgrade(&beacon_chain),
+            network_tx: network_send.clone(),
+            sync_tx: sync_send.clone(),
+            network_globals: network_globals.clone(),
+            executor: executor.clone(),
+            max_workers: cmp::max(1, num_cpus::get()),
+            current_workers: 0,
+            importing_blocks: Default::default(),
+            log: log.clone(),
+        }
+        .spawn_manager(beacon_processor_receive, None);
+
+        // generate the Message handler
+        let mut handler = Router {
+            network_globals,
+            chain: beacon_chain,
+            sync_send,
+            network: HandlerNetworkContext::new(network_send, log.clone()),
+            beacon_processor_send,
+            processor_log: log.new(o!("service" => "router")),
+            log: message_handler_log,
+        };
+
+        // spawn handler task and move the message handler instance into the spawned thread
+        executor.spawn(
+            async move {
+                debug!(log, "Network message router started");
+                UnboundedReceiverStream::new(handler_recv)
+                    .for_each(move |msg| future::ready(handler.handle_message(msg)))
+                    .await;
+            },
+            "router",
+        );
+
+        Ok(handler_send)
+    }
+
+    /// Handle all messages incoming from the network service.
+    fn handle_message(&mut self, message: RouterMessage<T::EthSpec>) {
+        match message {
+            // we have initiated a connection to a peer or the peer manager has requested a
+            // re-status
+            RouterMessage::PeerDialed(peer_id) | RouterMessage::StatusPeer(peer_id) => {
+                let status_message = status_message(&self.chain);
+                debug!(self.processor_log, "Sending Status Request"; "peer" => %peer_id, &status_message);
+                self.network
+                    .send_processor_request(peer_id, Request::Status(status_message));
```

Comment:
for consistency with the others, can we move this it's own `send_status`?

Reply:
I'm not sure if the the last commit I pushed was the right way to solve this, let me know if you were thinking in something else